src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -466,7 +466,11 @@ public ModelMetadata getMetadata(String modelId) {
         }
 
         private String getMapping() throws IOException {
-            URL url = ModelDao.class.getClassLoader().getResource(MODEL_INDEX_MAPPING_PATH);
+            URL url = null;
+            if (ModelDao.class.getClassLoader() != null) {
```

Comment:
Why would the class loader be null?

Reply:
Ideally not, except in the case of Bootstrap Class Loader. Not 100% sure if there is any scenario which can lead to it and might end up throwing null pointer exception. As it is called out in the findings I thought it is better to add a null validation check.
https://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Class.html#getClassLoader()

Reply:
This wont use the bootstrap class loader (https://docs.oracle.com/cd/E19501-01/819-3659/beadf/index.html). But, lets just add a null check and throw an exception instead of skipping

Follow this for below as well